### PR TITLE
MNT: Switch from flake8 to ruff for linting

### LINKED ISSUE #341: MNT: Change linting / precommit to use ruff
GeoPandas switched to ruff a while back; we should do the same.

### PR DESCRIPTION
**Pull request by brendan-ward**
Resolves #341 

Similar to changes in [GeoPandas #2921](https://github.com/geopandas/geopandas/pull/2921) but without needing to configure additional rules.

### CODE REVIEW

**Review by martinfleis — APPROVED** [review]
Thanks! We could potentially also consider switching from black to ruff-format. I am using it in some other projects and the transition was flawless.

**Comment by brendan-ward** [comment]
Switched to ruff-format and removed unnecessary `# noqa` statements.

**Comment by theroggy** [comment]
Maybe add `ruff` as dependency in environment-dev.yml?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
     hooks:
-      - id: black
-        language_version: python3
-  - repo: https://github.com/pycqa/flake8
-    rev: 5.0.4
-    hooks:
-      - id: flake8
-        language: python_venv
+      - id: ruff
+      - id: ruff-format
```
Comment (jorisvandenbossche) [inline .pre-commit-config.yaml:7]: Should we put the formatter first? (eg I can imagine that the linter complains about a too long line, which would already be solved by the formatter afterwards)

**Comment by jorisvandenbossche** [comment]
> Maybe add `ruff` as dependency in environment-dev.yml?

`pre-commit` is included, and then the idea is that you install the pre-commit, which will handle getting ruff (that's also how it went for black and flake8, which aren't included in the yml)

Of course if you don't use pre-commit and want to run ruff manually, you also need it in your env.

**Comment by theroggy** [comment]
> > Maybe add `ruff` as dependency in environment-dev.yml?
> 
> `pre-commit` is included, and then the idea is that you install the pre-commit, which will handle getting ruff (that's also how it went for black and flake8, which aren't included in the yml)
> 
> Of course if you don't use pre-commit and want to run ruff manually, you also need it in your env.

I personally use visual studio code and then it is convenient that formatter and linter are installed so the formatting is automatically applied and linting errors are shown. Once you commit pre-commit will check this as well... but this is later in the process.

But, if this is rather typical for my workflow, no problem not to include it... not problem to install it myself...

File: .pre-commit-config.yaml (hunk 1 above)
Comment (brendan-ward) [reply .pre-commit-config.yaml:7]: Sure; was just following the [example](https://github.com/astral-sh/ruff-pre-commit).

I'm assuming we don't want to auto-fix lint errors, which is where putting the linter first is recommended.

**Comment by brendan-ward** [comment]
Given that we had `black` in the `environment-dev.yml` (and I removed here), I have no issues with adding ruff as a development dependency.  For vscode, I've been using the [ruff extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff), which I believe ships with its own install of ruff that would be used to highlight things in the vscode UI.

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**